### PR TITLE
Aggregation expression refactor

### LINKED ISSUE #2652: Refactor aggregation expression code
Move `NewExpression` and related code to `handlers/common/aggregations`. Remove `Expression` interface. Think of better names for everything (given that "aggregations" is already in the package name).

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #2652.

Moving expressions to the `aggregations` package. Removing interface. Some smaller naming changes.

## Readiness checklist

* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by w84thesun — DISMISSED** [review]
LGTM!

File: internal/handlers/common/aggregations/expression.go [hunk 1]
```diff
@@ -12,31 +12,32 @@
...
 
-	// ErrEmptyFieldPath indicates that path is empty.
-	ErrEmptyFieldPath
+	// ErrInvalidExpression indicates that expression is invalid.
+	ErrInvalidExpression
 
-	// ErrInvalidFieldPath indicates that path is invalid.
-	ErrInvalidFieldPath
+	// ErrEmptyFieldPath indicates that field path expression is empty.
+	ErrEmptyFieldPath
```
Comment (chilagrow) [inline internal/handlers/common/aggregations/expression.go:40]:
```suggestion
	ErrEmptyExpression
```
To be more consistent with other namings, it looks like `$`.

File: internal/handlers/common/aggregations/expression.go [hunk 2]
```diff
@@ -154,20 +149,20 @@ func (p *pathExpression) Evaluate(doc *Document) any {
...
 
 	if len(vals) == 0 {
 		if isPrefixArray {
 			// when the prefix is array, return empty array.
-			return must.NotFail(NewArray())
+			return must.NotFail(types.NewArray())
 		}
 
-		return Null
+		return types.Null
```
Comment (chilagrow) [inline internal/handlers/common/aggregations/expression.go:165]: Can you add comment here please why we return Null? :hugs:

File: internal/handlers/common/aggregations/expression.go [hunk 3]
```diff
@@ -97,54 +92,54 @@ func NewExpressionWithOpts(expression string, opts *ExpressionOpts) (Expression,
...
-	path := p.path
+func (e *Expression) Evaluate(doc *types.Document) any {
+	path := e.path
 
 	if path.Len() == 1 {
 		val, err := doc.Get(path.String())
 		if err != nil {
 			// if the path does not exist, return nil.
-			return Null
+			return types.Null
```
Comment (chilagrow) [inline internal/handlers/common/aggregations/expression.go:142]: Could you replace the comment with why we return Null please?

File: internal/handlers/common/aggregations/expression.go [hunk 4]
```diff
@@ -45,36 +46,30 @@ const (
 	ErrEmptyVariable
 )
 
-// FieldPathError describes an error that occurs getting path from field.
-type FieldPathError struct {
+// ExpressionError describes an error that occurs getting path from field.
```
Comment (chilagrow) [inline internal/handlers/common/aggregations/expression.go:49]: The doc string needs updating

File: internal/handlers/common/aggregations/stages/group.go [hunk 5]
```diff
@@ -187,46 +187,47 @@ func (g *group) groupDocuments(ctx context.Context, in []*types.Document) ([]gro
...
 			)
-		case types.ErrEmptyVariable:
+		case aggregations.ErrEmptyVariable:
 			return nil, commonerrors.NewCommandErrorMsgWithArgument(
 				commonerrors.ErrFailedToParse,
 				"empty variable names are not allowed",
 				"$group (stage)",
 			)
-		case types.ErrUndefinedVariable:
+		case aggregations.ErrUndefinedVariable:
```
Comment (chilagrow) [inline internal/handlers/common/aggregations/stages/group.go:223]: Ah this is fine for `$$INVALID` which is a invalid expression, for valid expression like `$$NOW`, it should be unimplemented. Could you add this todo https://github.com/FerretDB/FerretDB/issues/2275 please.

**Review by chilagrow — COMMENTED** [review]
Great work, have a few comments :hugs:

File: internal/handlers/common/aggregations/expression.go (hunk 1 above)
Comment (noisersup) [reply internal/handlers/common/aggregations/expression.go:40]: I thought that using FieldPath here makes sense as the error is related to specific type of Expression and not all Expressions in general.

**Review by rumyantseva — APPROVED** [review]
The changes look good to me. Only one comment regarding documentation.

File: internal/handlers/common/aggregations/expression.go (hunk 3 above)
Comment (noisersup) [reply internal/handlers/common/aggregations/expression.go:143]: done

**Review by rumyantseva — APPROVED** [review]
LGTM

File: internal/handlers/common/aggregations/expression.go (hunk 1 above)
Comment (chilagrow) [reply internal/handlers/common/aggregations/expression.go:40]: yeah that's sensible.

**Review by chilagrow — APPROVED** [review]
Looks great!

**Pull request merged**